hack/make-rules/test.sh

```diff
@@ -250,22 +241,37 @@ runTests() {
   # must make sure the output from PARALLEL runs is not mixed. To achieve this,
   # we spawn a subshell for each PARALLEL process, redirecting the output to
   # separate files.
-
-  printf "%s\n" "${@}" \
-    | xargs -I{} -n 1 -P "${KUBE_COVERPROCS}" \
+  #
+  # Writing to the combined stdout is serialized with flock.
```

Comment:
Isn't the point of letting gotestsum control execution that it can handle this for us?

Reply:
That only works when `gotestsum` runs one `go test` command. Here the raw command starts multiple different `go test` commands in parallel because of `xargs -P "${KUBE_COVERPROCS}"`.

If we don't lock, then there can be arbitrary and broken interleaving of the different outputs.

According to the test.sh comment above, this workaround is necessary because "Go currently doesn't support collecting coverage across multiple packages at once". This made me wonder: does this limitation really still apply? No, it doesn't.

I'll use that to simplify test.sh.

Reply:
Done:
* https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/125534/pull-kubernetes-unit/1819021789792047104
* https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/125534/pull-kubernetes-integration/1819021789502640128

Reply:
But we can also do this:

/test pull-kubernetes-coverage-unit

Reply:
Works in the sense that it finds packages with low coverage and produces an image summarizing the coverage:

https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/125534/pull-kubernetes-coverage-unit/1819036751289651200